src/main/java/seedu/address/ui/StatsWindow.java

```diff
@@ -42,8 +44,17 @@ public StatsWindow() {
     /**
      * Replaces the current data used to generate the {@code PieChart} with the new data specified.
      */
+    //Solution adapted from: https://stackoverflow.com/questions/36276805/how-to-wrap-text-of-a-javafx-chart-legend
     public void setPieChart(ObservableList<PieChart.Data> newData) {
         chart.setData(newData);
+        for (Node node : chart.lookupAll(".chart-legend-item")) {
+            if (node instanceof Label) {
+                Label n = (Label) node;
```

Comment:
rename n to legendElement or something similar? maybe put a comment as to what this does as well which is to wrap the text in the legend